pymongo/synchronous/mongo_client.py

```diff
@@ -2773,14 +2776,19 @@ def run(self) -> T:
                 # most likely be a waste of time.
                 raise
             except PyMongoError as exc:
+                overloaded = False
                 # Execute specialized catch on read
                 if self._is_read:
                     if isinstance(exc, (ConnectionFailure, OperationFailure)):
                         # ConnectionFailures do not supply a code property
                         exc_code = getattr(exc, "code", None)
-                        if self._is_not_eligible_for_retry() or (
-                            isinstance(exc, OperationFailure)
-                            and exc_code not in helpers_shared._RETRYABLE_ERROR_CODES
+                        overloaded = exc.has_error_label("Retryable")
```

Comment:
Wouldn't it be more appropriate to mark overloaded when the error label `SystemOverloadedError` is present? See [this line](https://github.com/10gen/mongo/blob/8536c690f50947f2fbc41049d00fe64e39e982c3/src/mongo/db/error_labels.h#L55).Today our rate limiter is the only place where we use `Retryable` error label, but there might be more use cases in the future. I think `SystemOverloadedError` would convey the meaning better.

Reply:
Ah! Yes. We do add this error label to provide a way for the server to report operations as being idempotent. The `SystemOverloadedError` was actually added in [SPM-3784](https://jira.mongodb.org/browse/SPM-3784) and implemented in the [SERVER-96975](https://jira.mongodb.org/browse/SERVER-96975) ticket.

> So the correct logic is to retry IFF "Retryable" is present and apply backoff IFF "SystemOverloadedError" is present?

Yes. In the design document of SPM-4003, it defines both DefaultRetryStrategy and AdaptiveRetryStrategy to use SystemOverloadedError for overloading specific behaviour. The request rate limiter (not to be confused with the ingress connection rate limiter) already return an error with this label.

Reply:
Thanks. Fixed.